chain-test-utils/src/time/mod.rs

```diff
@@ -0,0 +1,136 @@
+use chain_time::{Epoch, Slot, TimeEra};
+use smoke::{Generator, R};
+
+// Generate an `Epoch` from a generator function
+fn generate_epoch_with<GenF>(mut gen: GenF) -> Epoch
+where
+    GenF: FnMut() -> u32,
+{
+    Epoch(gen())
+}
+
+// Generate an `Slot` from a generator function
+fn generate_slot_with<GenF>(mut gen: GenF) -> Slot
+where
+    GenF: FnMut() -> u64,
+{
+    gen().into()
+}
+
+// Generate an `Epoch` given a `smoke::R` (random generator)
+pub fn generate_epoch(r: &mut R) -> Epoch {
+    generate_epoch_with(|| r.num())
+}
+
+// Generate an `Epoch` given a `smoke::R` (random generator) and a `(u32, u32)` range limit tuple
+pub fn generate_epoch_with_range(r: &mut R, range: (u32, u32)) -> Epoch {
+    generate_epoch_with(|| r.num_range(range.0, range.1))
+}
+
+// Generate an `Slot` given a `smoke::R` (random generator)
+pub fn generate_slot(r: &mut R) -> Slot {
+    generate_slot_with(|| r.num())
+}
+
+// Generate an `Slot` given a `smoke::R` (random generator) and a `(u32, u32)` range limit tuple
+pub fn generate_slot_with_range(r: &mut R, range: (u64, u64)) -> Slot {
+    generate_slot_with(|| r.num_range(range.0, range.1))
+}
+
+// Generate an `TimeEra` given a `smoke::R` (random generator)
+pub fn generate_time_era(r: &mut R) -> TimeEra {
+    TimeEra::new(generate_slot(r), generate_epoch(r), r.num())
+}
+
+// `TimeEra` configuration, encapsulates the building boundaries for the inner data
+#[derive(Clone)]
+pub struct TimeEraGenCfg {
+    pub slot_range: (u64, u64),
+    pub epoch_range: (u32, u32),
+    pub slots_per_epoch_range: (u32, u32),
+}
+
+// Generate an `TimeEra` given a `smoke::R` (random generator) and a `TimeEraGenCfg` range limit tuple
+pub fn generate_time_era_with_config(r: &mut R, config: &TimeEraGenCfg) -> TimeEra {
```

Comment:
This (as a public API) also looks redundant to the composition of `TimeEraGenerator::with_config` and the `gen` method.

Reply:
Well, my idea was that some times you will need a one shot time build and some times you will need a bunch of generated items. It feels overkill to build a generator just to get one object when you can just call a method.